src/java/org/apache/cassandra/locator/DynamicEndpointSnitch.java

```diff
@@ -188,6 +343,9 @@ public String getDatacenter(InetAddressAndPort endpoint)
     public <C extends ReplicaCollection<? extends C>> C sortedByProximity(final InetAddressAndPort address, C unsortedAddresses)
     {
         assert address.equals(FBUtilities.getBroadcastAddressAndPort()); // we only know about ourself
+        for (Replica unsortedAddress : unsortedAddresses)
```

Comment:
We shouldn't need to fetch each endpoint from the map a second time.

Reply:
So I can easily eliminate the extra iteration when calling into `sortedByProximityWithBadness`, but `sortedByProximityWithScore` only calls `compareEndpoints` and putting two more dict lookups there seemed likely slower (to me) than just iterating the unsortedAddresses in `sortedByProximity`. I can make this method more complicated to special case `sortedByProximity -> sortedByProximityWithScore` (as opposed to the normal path of `sortedByProximity` -> sortedByProximityWithBadness -> sortedByProximityWithScore`) ... sure

Reply:
Hmm... yeah... seems like the score maybe be merged with the other map containing the reservoir and the score is just an attribute of the annotated thing.

Reply:
I don't think that combining the two maps is a good idea and this loop seems pretty cheap to me whether we do it here first or add a statement to a later loop that reads the score map. Due to the way that we alias scores to and do full copy on write I'm also worried about correctness given the number of comments explaining why that scores aliasing is so important I don't want to risk changing that in this patch.

I do agree a through examination of the scores datastructure and way that we update it may be warranted in the future (e.g. we could create some kind of immutable view just for the endpoints we're ranking, although I don't know a way to do that without adding lots of extra garbage).